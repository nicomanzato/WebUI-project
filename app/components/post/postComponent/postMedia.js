import React, { Component } from 'react';
import {
  StyleSheet,
  View,
  Text,
  Image,
  TouchableHighlight
} from 'react-native';

export const PostMedia = (props) => {

  const renderMedia = (media) => {
    let output;
    if (media) {
      output = media.map((image) => {
        return(
          <Image
            source={{uri: image.media_url_https}}
            style={styles.postImage}
            fadeDuration={0}
            key={image.id}/>
        );
      });
    }

    return output;
  }

  return (
    <View>
      { renderMedia(props.post.entities.media) }
    </View>
  );
}

const styles = StyleSheet.create({

  postImage: {
    borderRadius: 10,
    resizeMode: 'contain',
    height: 250
  },

});
