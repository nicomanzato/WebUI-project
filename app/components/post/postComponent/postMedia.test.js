import React from 'react'
import {PostMedia} from './postMedia'
import PostMock2 from './../../../store/post/mock/postMock2'
import { shallow, mount, ReactWrapper } from 'enzyme'

describe('post component', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <PostMedia post={PostMock2} />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
