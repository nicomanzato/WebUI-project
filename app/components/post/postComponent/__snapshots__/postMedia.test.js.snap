// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component should render correctly 1`] = `
<Component>
  <Image
    fadeDuration={0}
    key="1076097964253298700"
    source={
      Object {
        "uri": "https://pbs.twimg.com/media/Du8RZabWwAA5fiX.jpg",
      }
    }
    style={
      Object {
        "borderRadius": 10,
        "height": 250,
        "resizeMode": "contain",
      }
    }
  />
</Component>
`;
